fix golangci.yml and some warnings

Vitomir2 opened the pull request:
# Description

- update go version to 1.21 and use `go mod tidy`;
- fix `golangci.yml` and some warnings related to property changes;
- add other paths for exclusion rules and make the line length 120 and add the output configuration;
- fix and add the lint, ci, build, and test workflows;
- fetch-depth is used to not doing a shallow checkout which avoids diff error for the linked revision;
- remove the sonar token to avoid the usage of the sonar's analyses;
- remove the race and shuffle flags from the make test command;
- optimize the `TestThrottling` test;
- set a separate `statusCh` variable to store the peer client and remove the `t.Skip()` from the `TestStatusPubSub` test;
- optimize the codebase based on the golintci issues;
- change `DefaultChainID` to `8844`;
- change error string to start with small letter in order to follow the golang conventions;
- actions/setup-go must be done after actions/setup-checkout for cache optimizations.
- updated the version of the actions;
- add submodules: recursive in the checkout action;

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [ ] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [x] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

Vitomir2 committed:
fix golangci.yml and some warnings

Vitomir2 committed:
fix linting and modify golangci-lint config

skip some linters and fix errors where suitable to;
set DefaultChainID to 8844 and move it in the polybft package in order to reuse it in multiple packages;

Vitomir2 committed:
move some paused workflows and test

Vitomir2 committed:
update build.yml to build properly and upload artifacts

Vitomir2 committed:
update build command

Vitomir2 committed:
activate the go_build_reproducibility

Vitomir2 committed:
add the test.yml to the workflows and remove the sonar token

Vitomir2 committed:
update test workflow

rename the go_test name;
change the sprintf string to accept a string instead of a number for the r.Amount;

Vitomir2 committed:
make a separate variable for peerStatusCh in order to have consistent channel usage

Vitomir2 committed:
change retention days

R-Santev commented on .golangci.yml:
```diff
@@ -3,14 +3,6 @@
-  #   vendor$, third_party$, testdata$, examples$, Godeps$, builtin$
-  skip-dirs-use-default: true
-  skip-dirs:
-    - tests
```
why the skip flags are removed

Vitomir2 replied:
@R-Santev, the `Golangci-lint` warned me that these flags are not allowed anymore in the yml configuration. Therefore:

1. skip-dirs-use-default is always true by default
2. skip-dirs is used as:
```
exclude-dirs:
    - tests
```
3. the service property is also not allowed and there was no difference running it w/o this property.

R-Santev commented on consensus/polybft/blockchain_wrapper.go:
```diff
@@ -224,7 +224,12 @@ func (p *blockchainWrapper) UnubscribeEvents(subscription blockchain.Subscriptio
 func (p *blockchainWrapper) GetChainID() uint64 {
-	return uint64(p.blockchain.Config().ChainID)
+	chainID := p.blockchain.Config().ChainID
+	if chainID < 0 {
```
better exclude the rule - chainId is always > 0

Vitomir2 replied:
Then, I will have to use the `//no-lint` tag. Is that okay? The linter warns us that there is a possibility of error in the conversion.

R-Santev commented on consensus/polybft/polybft.go:
```diff
@@ -30,9 +30,10 @@ import (
+	minSyncPeers   = 2
+	pbftProto      = "/pbft/0.2"
+	bridgeProto    = "/bridge/0.2"
+	DefaultChainID = 8844
```
the old place for this constant is better because chain ID is not polybft-specific term

Vitomir2 committed:
updates

revert the changes in the GetChainID;
move the default chain id in the default values of the command package;
remove the excluded rules;

ldez committed:
chore: fix linting (#72)

* chore: fix linting
* fix: invalid nolint directives
* fix: actions/upload-artifact

Vitomir2 committed:
updates

improve the event handlers and the waiting periods, avoid race conditions and optimize the tests codebase;
optimize the TestEventSubscription_ProcessedEvents, TestPeerConnectionUpdateEventCh and 'TestSimpleGossip' tests;
add TestTimeout and WaitFor helpers in the testing.go;

Vitomir2 committed:
revert

Vitomir2 committed:
optimize throttling test

Vitomir2 committed:
remove race and shuffle flags from make test cmd

Vitomir2 committed:
merged the content of the cleaner branch while preserving history

Vitomir2 committed:
update the github workflows and the golangci

Vitomir2 committed:
increase waiting time for the event

Vitomir2 committed:
revert the big.Int conversion and add //nolint

Vitomir2 committed:
fix

Vitomir2 committed:
fix

Vitomir2 committed:
rename
